### ISSUE TITLE
Why not transfer down current account?

### ISSUE THREAD

**Issue by blackmonkey** [evennia/evennia#2163]
I was trying to initialize some Rooms when create my customized Object:

```python
def at_object_creation(self):
    portal, err = Room.create("Portal", self.account)
```

However, I found `self.account` is `None` here, while the current account can be fetched from `request.user` in the previous call stack:

https://github.com/evennia/evennia/blob/bd656417552089a79dd33053266785b3517ff5ee/evennia/objects/admin.py#L168

I think it is better to save the account into `obj`, or transfer it to `obj.at_object_creation()`. I.e.
```python
obj.save()
if not change:
    obj.account = request.user
    obj.set_class_from_typeclass(typeclass_path=obj.db_typeclass_path)
    obj.basetype_setup()
    obj.basetype_posthook_setup()
    obj.at_object_creation()
obj.at_init()
```
or
```python
obj.save()
if not change:
    obj.set_class_from_typeclass(typeclass_path=obj.db_typeclass_path)
    obj.basetype_setup()
    obj.basetype_posthook_setup()
    obj.at_object_creation(request.user)
obj.at_init()
```

**Comment by strikaco** [comment]
`at_object_creation` is not always called with a `request` object available for its context though. You only have access to `request.user` because you created an object using the web interface. 

That's why you're expected to pass the account manually to the `create` method-- otherwise the API would be inconsistent between the web and the CLI; we can't presume that you'll always have a `request`, `session` or want to reference any account at all.

(You'll also run into the same problem when trying to do something seemingly simple like "grab the IP address." Is it in a `request` object or `session`? Which `session` in particular? Etc.)

**Comment by blackmonkey** [comment]
Then how about the first walkaround?
```python
obj.save()
if not change:
    obj.account = request.user
    obj.set_class_from_typeclass(typeclass_path=obj.db_typeclass_path)
    obj.basetype_setup()
    obj.basetype_posthook_setup()
    obj.at_object_creation()
obj.at_init()
```
If this is not ok, please fix issue#2162. Thanks in advance!

**Comment by blackmonkey** [comment]
> Then how about the first walkaround?
> 
> ```python
> obj.save()
> if not change:
>     obj.account = request.user
>     obj.set_class_from_typeclass(typeclass_path=obj.db_typeclass_path)
>     obj.basetype_setup()
>     obj.basetype_posthook_setup()
>     obj.at_object_creation()
> obj.at_init()
> ```
> 
> If this is not ok, please fix issue#2162. Thanks in advance!

I found that `obj.account = request.user` will mark the object as a Character no matter which type it actually is. So this walkaround is not reasonable. Please fix issue#2162. I'll close this one.
